packages/suite/src/reducers/wallet/discoveryReducer.ts

```diff
@@ -81,3 +82,25 @@ const discoveryReducer = (state: State = initialState, action: WalletAction | Su
     });
 
 export default discoveryReducer;
+
+type RootState = Pick<AppState, 'wallet' | 'suite'>;
+
+// Get discovery process for deviceState.
+export const selectDiscovery = (state: RootState, deviceState: string | undefined) =>
+    deviceState ? state.wallet.discovery.find(d => d.deviceState === deviceState) : undefined;
+
+export const selectDiscoveryForDevice = (state: RootState) =>
+    selectDiscovery(state, state.suite.device?.state);
+
+/**
+ * Helper selector called from components
+ * return `true` if discovery process is running/completed and `authConfirm` is required
+ */
+export const selectDiscoveryAuthConfirmation = createSelector(
```

Comment:
nit: I would suggest prefixing with `is` => `selectIsDiscoveryAuthConfirmation` and maybe even add `required` at the end, which will cause it to be quite longer, but in the other hand it will be super intuitive what it do

Reply:
Renamed and squashed, waiting for checks.